opsdroid/logging.py

```diff
@@ -60,71 +61,92 @@ def filter(self, record):
+    Args:
+        config: contains only the logging section of the configuration since we
+            don't care about anything else for logs.
 
-    rootlogger.setLevel(log_level)
+    """
+    formatter_str = "%(levelname)s %(name)s"
 
-    formatter_str = "%(levelname)s %(name)s:"
+    if config.get("formatter"):
+        return config["formatter"]
 
-    with contextlib.suppress(KeyError):
-        if config["logging"]["timestamp"]:
-            formatter_str = "%(asctime)s " + formatter_str
+    if config.get("extended"):
+        formatter_str += ".%(funcName)s():"
 
-    with contextlib.suppress(KeyError):
-        if config["logging"]["extended"]:
-            formatter_str = formatter_str[:-1] + ".%(funcName)s():"
+    if config.get("timestamp"):
+        formatter_str = "%(asctime)s " + formatter_str
 
     formatter_str += " %(message)s"
-    formatter = logging.Formatter(formatter_str)
 
-    console_handler = logging.StreamHandler()
-    console_handler.setLevel(log_level)
-    console_handler.setFormatter(formatter)
+    return formatter_str
 
-    with contextlib.suppress(KeyError):
-        console_handler.addFilter(ParsingFilter(config, config["logging"]["filter"]))
 
-    rootlogger.addHandler(console_handler)
+def configure_logging(config):
+    """Configure the root logger based on user config."""
+    rootlogger = logging.getLogger()
+    while rootlogger.handlers:
+        rootlogger.handlers.pop()
 
-    with contextlib.suppress(KeyError):
-        if not config["logging"]["console"]:
-            console_handler.setLevel(logging.CRITICAL)
+    try:
+        if config["path"]:
+            logfile_path = os.path.expanduser(config["path"])
+        else:
+            logfile_path = config["path"]
+    except KeyError:
+        logfile_path = DEFAULT_LOG_FILENAME
 
     if logfile_path:
         logdir = os.path.dirname(os.path.realpath(logfile_path))
         if not os.path.isdir(logdir):
             os.makedirs(logdir)
 
-        file_handler = RotatingFileHandler(
-            logfile_path, maxBytes=logging_config.get("file-size", 50e6)
+    log_level = get_logging_level(config.get("level", "info"))
+    rootlogger.setLevel(log_level)
+    formatter_str = set_formatter_string(config)
+    formatter = logging.Formatter(formatter_str)
+    handler = None
+
+    if config.get("rich") is not False:
+        handler = RichHandler(
+            rich_tracebacks=True,
+            show_time=config.get("timestamp", True),
+            show_path=config.get("extended", True),
         )
 
+    if logfile_path:
+        file_handler = RotatingFileHandler(
+            logfile_path, maxBytes=config.get("file-size", 50e6)
+        )
         file_handler.setLevel(log_level)
         file_handler.setFormatter(formatter)
+        rootlogger.addHandler(file_handler)
 
-        with contextlib.suppress(KeyError):
-            file_handler.addFilter(ParsingFilter(config, config["logging"]["filter"]))
+    if config.get("console"):
```

Comment:
So, given these configs will rich logging be enabled?
```
{"path": False, "console": True, "rich": True}
```
```
{"path": False, "console": True}
```

Reply:
We are using a single handler. The default is rich, but if you set both console and rich to true then you get console. Since both Rich and Console are logging to the terminal I don't see a point in setting both to True 

The second example, you will get console

Reply:
Ok. Can we add a warning if both are enabled explaining that console takes precedence?